### PR TITLE
First steps for downloadin config files into the recovery system.

### PR DESCRIPTION
**Pull request by jsmeix**
In general see https://github.com/rear/rear/issues/841

For the particular state see
https://github.com/rear/rear/issues/841#issuecomment-229069740

### CODE REVIEW

**Comment by jsmeix** [comment]
@gdha 
please have a look and
if it is o.k. for you, please merge it.

It is a pure additional functionality so that
it cannot cause backward incompatible regressions.

**Comment by jsmeix** [comment]
Got no objections and I cannot imagine what could break
but I think it is a first step that could help a lot
so that I merge it now.

File: usr/share/rear/init/default/03_update_recovery_system.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Tell the user that the recovery system will be updated:
+LogPrint "Updating recovery system with the content from '$RECOVERY_UPDATE_URL':"
+
+# Download the tar.gz that contains the update files.
+local update_archive_filename="recovery-update.tar.gz"
+# "curl -f" does not fail reliable (there are occasions where non-successful response codes will slip through)
+# therefore the HTTP response code is written to stdout ('-w') so that it can be explicitly tested afterwards
+# ('-s' does not show progress meter or error messages but 'S' it makes it show an error message if it fails)
+# "curl --verbose" messages go to stderr so that they go to the "rear recover" log file:
+local http_response_code=$( curl $verbose -f -s -S -w "%{http_code}" -o /$update_archive_filename $RECOVERY_UPDATE_URL )
```
Comment (gdha) [inline usr/share/rear/init/default/03_update_recovery_system.sh]: @jsmeix just make sure that `curl` is added to the `REQUIRED_PROGS` array if we depend on it in recovery mode.

**Comment by jsmeix** [comment]
Unfortunately in the Git commit message
I used the outdated config variable name
RECOVERY_CONFIG_URL.
The right name is RECOVERY_UPDATE_URL.
See default.conf how RECOVERY_UPDATE_URL works.

File: usr/share/rear/init/default/03_update_recovery_system.sh (hunk 1 above)
Comment (jsmeix) [inline usr/share/rear/init/default/03_update_recovery_system.sh]: @gdha 
in my current first step state one must do that manually via
<pre>
REQUIRED_PROGS=( "${REQUIRED_PROGS[@]}" curl )
</pre>
see my description in default.conf.
The new RECOVERY_UPDATE_URL functionality
will require further enhancements and then - as needed
by the usres - I will try to make things work more
automatically right out of the box.

**Comment by jsmeix** [comment]
For a possible use case, see https://github.com/rear/rear/issues/943
in particular see https://github.com/rear/rear/issues/943#issuecomment-236547810

**Pull request merged**